tests/onnx/test_onnx_v2.py

```diff
@@ -191,6 +191,7 @@ def test_values_override(self):
     ("deberta", "microsoft/deberta-base"),
     ("deberta-v2", "microsoft/deberta-v2-xlarge"),
     ("convnext", "facebook/convnext-tiny-224"),
+    ("decision_transformer", "huggingface/decision_transformer"),
```

Comment:
```suggestion
    ("decision-transformer", "edbeeching/decision-transformer-gym-hopper-medium"),
```
It seems this checkpoint does not exist in the Hugging Face Hub. I suggest this one.

Reply:
@regisss done, I modified test_onnx_v2 to point to the correct decision-transformer model as you identified above, also listed below is the output of the command you specified

saikatkanjilal@system76-pc:~/code/opensource/transformers$ RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -k "decision-transformer"
========================================================================================= test session starts =========================================================================================
platform linux -- Python 3.10.0rc1, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/saikatkanjilal/code/opensource/transformers, configfile: setup.cfg
collected 432 items / 432 deselected / 0 selected                                                                                                                                                     

========================================================================================== warnings summary ===========================================================================================
src/transformers/testing_utils.py:28
  /home/saikatkanjilal/code/opensource/transformers/src/transformers/testing_utils.py:28: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
    from distutils.util import strtobool

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================= 432 deselected, 1 warning in 0.30s ==================================================================================


As you can see I just received warnings but no errors, are we close?

Reply:
@skanjila The test was not run because it could not find it. It happens because in `test_onnx_v2.py` you did not change the first element of the tuple to `decision-transformer` (`-` instead of `_` is a convention we use for model names). Could you change this and re-run the test command please?

Reply:
@regisss I get this error when I put dash instead of underscore:  E   KeyError: "decision-transformer is not supported yet. Only ['albert', 'bart', 'beit', 'bert', 'big-bird', 'bigbird-pegasus', 'blenderbot', 'blenderbot-small', 'bloom', 'camembert', 'clip', 'codegen', 'convbert', 'convnext', 'data2vec-text', 'data2vec-vision', 'deberta', 'deberta-v2', 'decision_transformer', 'deit', 'detr', 'distilbert', 'electra', 'flaubert', 'gpt2', 'gptj', 'gpt-neo', 'groupvit', 'ibert', 'layoutlm', 'layoutlmv3', 'levit', 'longt5', 'longformer', 'marian', 'mbart', 'mobilebert', 'mobilevit', 'mt5', 'm2m-100', 'owlvit', 'perceiver', 'resnet', 'roberta', 'roformer', 'segformer', 'squeezebert', 't5', 'vit', 'xlm', 'xlm-roberta', 'yolos'] are supported. If you want to support decision-transformer please propose a PR or open up an issue."


Note that the directory for decision transformer under model has an underscore

Reply:
@regisss if you look at the error message it's actually looking for an underscore, did I miss something here?

Reply:
@skanjila  Ah you also need to put the dash in the `features.py` file, I missed that

Reply:
@regisss I added the dash in features.py as shown here and in the latest commit :  "decision-transformer": supported_features_mapping(
            "default",
            onnx_config_cls="models.decision_transformer.DecisionTransformerOnnxConfig",
        ),  , listed below are the results:
saikatkanjilal@system76-pc:~/code/opensource/transformers$ RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -k "decision-transformer"
========================================================================================= test session starts =========================================================================================
platform linux -- Python 3.10.0rc1, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/saikatkanjilal/code/opensource/transformers, configfile: setup.cfg
collected 432 items / 432 deselected / 0 selected                                                                                                                                                     

========================================================================================== warnings summary ===========================================================================================
src/transformers/testing_utils.py:28
  /home/saikatkanjilal/code/opensource/transformers/src/transformers/testing_utils.py:28: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
    from distutils.util import strtobool

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================= 432 deselected, 1 warning in 0.30s ==================================================================================


Let me know next steps

Reply:
@skanjila Sorry it should be an underscore in the pytest command actually: `RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -k "decision_transformer"`.

I just ran it on my laptop and it fails because the default method for generating dummy inputs to export the model to the ONNX format expects a preprocessor but got none. Which makes sense according to the configuration of the model.
I don't know this model at all but I think you will have to override the [`generate_dummy_inputs`](https://github.com/huggingface/transformers/blob/c875a96eb154f2ea4a831a48b47c37c1c3da99e7/src/transformers/onnx/config.py#L264) method in `DecisionTransformerOnnxConfig` to adapt it for this model. According to the [documentation](https://huggingface.co/docs/transformers/v4.22.2/en/model_doc/decision_transformer#transformers.DecisionTransformerModel.forward), the inputs to the `forward` method of the model are states, actions and rewards. So you will have to generate dummy states, dummy actions and dummy rewards.

Reply:
@regisss sigh sorry I feel like we're lost about 10 steps of progress based on the above work that still needs to be done, the above is going to take a bit of time, let me take a crack at this and I will make the changes to the PR and I may need help on the above

Reply:
@skanjila  Yeah I know I didn't anticipate this since I've never used this model :confused: 

I can definitely help with this. Have you already used this model within the Transformers library? Because if you're familiar with the inputs it takes it could be quite straightforward.

Reply:
@regisss no I have not, give me a bit of time to learn this, thats the whole point of me contributing anyway, i will ping you through the journey